core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/tracer/impl/CurrentSpanNameTracingCondition.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.internal.profiling.tracing.event.tracer.impl;
+
+import org.mule.runtime.core.internal.profiling.tracing.event.span.InternalSpan;
+import org.mule.runtime.core.internal.profiling.tracing.event.tracer.TracingCondition;
+import org.mule.runtime.core.internal.profiling.tracing.event.tracer.TracingConditionNotMetException;
+
+/**
+ * A {@link TracingCondition} that verifies that the current span has a certain name.
+ *
+ * @since 4.5.0
+ */
+public class CurrentSpanNameTracingCondition implements TracingCondition {
+
+  private final String currentSpanExpectedName;
+
+  public CurrentSpanNameTracingCondition(String currentSpanExpectedName) {
+    this.currentSpanExpectedName = currentSpanExpectedName;
+  }
+
+  @Override
+  public void assertOnCurrentSpan(InternalSpan currentSpan) throws TracingConditionNotMetException {
+    if (currentSpanExpectedName.equals("mule:flow:route")) {
```

Comment:
Is this really needed? If it is related to my current task, can we add the TODO?

Reply:
Opps. No. I think it was left there because of a test. That shouldn't be neded.